src/components/colorscale/defaults.js

```diff
@@ -43,7 +42,15 @@ module.exports = function colorScaleDefaults(traceIn, traceOut, layout, coerce,
     var autoColorscaleDflt;
     if(sclIn !== undefined) autoColorscaleDflt = !isValidScale(sclIn);
     coerce(prefix + 'autocolorscale', autoColorscaleDflt);
-    var sclOut = coerce(prefix + 'colorscale');
+
+    var layoutColorscale = layout.colorscale || {};
+    containerOut.diverging = layoutColorscale.diverging;
+    containerOut.sequential = layoutColorscale.sequential;
+    containerOut.sequentialminus = layoutColorscale.sequentialminus;
+    var dfltScl = containerOut.diverging;
+    var sclOut;
+    if(dfltScl) sclOut = coerce(prefix + 'colorscale', dfltScl);
```

Comment:
Ha I see. We wouldn't need this extra `if-else` block if you made the `layout.colorscale.*.dflt` `undefined` (or removed them entirely) instead of `false`. The relevant line in `coerce.js` is:

https://github.com/plotly/plotly.js/blob/695f3114c86fee146c7f8e4b8700fa0b00d5c304/src/lib/coerce.js#L384

Reply:
... But looking at this again, do we really need logic in colorscale/defaults.js **and** in `colorscale/calc.js`?

Ideally, 

- I would leave colorscale/defaults.js untouched
- make the `layout.colorscale.*.dflt` match `scales.RdBu`, `scales.Reds` and `scales.Blues`, making those `container.diverging || scales.RdBu;` in colorscale/calc.js unnecessary.
- handle the autocolorscale logic in colorscale/calc.js

Now, I'm just thinking out-loud. @antoinerg let me know if my proposal has any flaws.

Reply:
I think this makes a lot of sense. Let me rework this PR!